test/Infrastructure/LeanCode.AuditLogs.Tests/AuditLogsMiddlewareTests.cs

```diff
@@ -0,0 +1,204 @@
+using System.Diagnostics;
+using System.Text.Encodings.Web;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using FluentAssertions;
+using LeanCode.CQRS.MassTransitRelay;
+using LeanCode.OpenTelemetry;
+using MassTransit;
+using MassTransit.Testing;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.TestHost;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using OpenTelemetry.Resources;
+using OpenTelemetry.Trace;
+using Xunit;
+using static LeanCode.AuditLogs.Tests.AuditLogsFilterTests;
+
+namespace LeanCode.AuditLogs.Tests;
+
+public sealed class AuditLogsMiddlewareTests : IAsyncLifetime, IDisposable
```

Comment:
I totally don't get why we need MassTransit here. We're testing `AuditLogsMiddleware` that is unrelated to MT - it only publishes messages (and we can assert that differently - we can have mocks for that).

I also don't think we need WebHost here at all. We don't really rely on HTTP in the middleware, so there is no point in putting everything just to test two lines of code.

Reply:
Also, if you extract the "gather what changed and publish message" logic into a separate class, then test it independently of MT/WebHost, then testing both filter & middleware assuming that it works will be _much_ simpler.